langserver/handler.go

```diff
@@ -556,7 +556,8 @@ func (h *langHandler) lint(ctx context.Context, uri DocumentURI, eventType event
 					continue
 				}
 			} else {
-				if diagURI != uri && !config.LintWorkspace {
+				diagFname, _ := fromURI(diagURI)
```

Comment:
It is safe to ignore error here because `diagURI` is either `uri` (which already was checked for `fromURI()` error) or result of `toURI()`).